vdr/didstore/store.go

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright (C) 2022 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package didstore
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/go-stoabs"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/storage"
+	vdr "github.com/nuts-foundation/nuts-node/vdr/types"
+)
+
+const (
+	// latestShelf has DID as key and latest documentMetadata reference as value.
+	// shelfs have a V2 postfix due to overlapping names with previous implementation
+	latestShelf = "latestV2"
```

Comment:
Why not migrate?

Reply:
Migration is started at `start` of the VDR. It uses the network reprocess to reevaluate all DID Document transactions. This will automatically fix any broken situation and fill diagnostics. Migration from existing data is not possible since it's missing transactional (ordering) data.

As a cleanup, old shelves may be deleted, but stoabs can't do that yet. Will make an issue.

Reply:
#1785